### PR TITLE
fix(entry): added idle loop break;

### PR DESCRIPTION
**Pull request by ThornWalli**
## Entry
- Idle loop termination added. Number is defined with `MAX_IDLE_TRYS`.
- New activation function for element `#nuxt-speedkit-button-init-performance-issue` -> `initPerformanceIssue`.
- Revision of dom `id` passing `updateInfoLayerMessage` & `observeSpeedkitButton`.

## SpeedkitLayer

- Added new layer message `nuxt-speedkit-message-performance-issue`.
- "Font only" action has been removed, it is enough to choose between:
  1. page will be executed normally (`#nuxt-speedkit-button-init-app`)
  2. or scripts are not initialized.  (`#nuxt-speedkit-button-init-performance-issue`)
      - Sets the CSS class `nuxt-speedkit-performance-issue` on the `html` tag.
      - Activates all fonts by setting the class `font-active` on all elements with the attribute `data-font`.
      - Converts all not activated pictures (`:hydrate="false"`) from `noscript` to `picture`.

### CODE REVIEW

File: lib/templates/entry.js [hunk 1]
```diff
@@ -1,30 +1,41 @@
 import { hasSufficientPerformance, hasSufficientDownloadPerformance, setup } from 'nuxt-speedkit/utils/performance';
 import { isSupportedBrowser } from 'nuxt-speedkit/utils/browser';
 
+const MAX_IDLE_TRYS = 1000;
```
Comment (kaczmen) [inline lib/templates/entry.js:4]: Bad spelling, should be `MAX_IDLE_TRIES`.

File: lib/templates/entry.js [hunk 2]
```diff
@@ -1,30 +1,41 @@
...
-  return import(<% if (!options.ssr) { %>/* webpackMode: "eager" */<% } %>'../client');
+  init = force || await new Promise(resolve => waitForIdle(resolve));
+  if (init) {
+    return import(<% if (!options.ssr) { %>/* webpackMode: "eager" */<% } %>'../client');
+  }
+
+  return null;
 };
 
+let idleTrys = 0;
```
Comment (kaczmen) [inline lib/templates/entry.js:21]: As before, should be spelled `idleTries`.

**Review by kaczmen — COMMENTED** [review]
Spelling mistakes.

**Comment by kaczmen** [comment]
`requestIdleCallback` infinite loop problem is a serious bug. Should be hotfixed ASAP with production release.

**Review by StephanGerbeth — APPROVED** [review]

**Comment by ThornWalli** [comment]
@kaczmen PR is now in the `beta` and can be included via beta npm dependency if it should go fast.

Otherwise wait a bit, will come today or tomorrow in the `main` branch.

**Comment by Biincle** [comment]
When can i expect fix with loop problem at production release?

**Comment by ThornWalli** [comment]
Hello @Biincle,

I will put another smaller change in the `beta`.
Concerns option `maxIdleTries`, this will be changed to `maxIdleDuration`. 
It will then be defined by seconds. Default: `15s` analog to current `maxIdleTries` Option.

After that it will be in production.

**Comment by ThornWalli** [comment]
@Biincle @kaczmen 

Is in production now 😉  (#605)

If the `SpeedkitLayer` is used, this could be interesting https://github.com/GrabarzUndPartner/nuxt-speedkit/pull/607

**Pull request merged**